public/app/features/explore/Logs/Logs.tsx

```diff
@@ -671,39 +691,51 @@ const UnthemedLogs: React.FunctionComponent<Props> = (props: Props) => {
     }
 
     // find the Logs parent item
-    const logsParent = outlineItems?.find((item) => item.panelId === 'Logs' && item.level === 'root');
+    const logsParent = outlineItems?.find((item) => item.panelId === PINNED_LOGS_PANELID && item.level === 'root');
 
     //update the parent's expanded state
     if (logsParent && updateItem) {
       updateItem(logsParent.id, { expanded: true });
     }
 
-    register?.({
-      icon: 'gf-logs',
-      title: 'Pinned log',
-      panelId: 'Logs',
-      level: 'child',
-      ref: null,
-      color: LogLevelColor[row.logLevel],
-      childOnTop: true,
-      onClick: () => {
-        onOpenContext(row, () => {});
-        reportInteraction('explore_toolbar_contentoutline_clicked', {
-          item: 'section',
-          type: 'Logs:pinned:pinned-log-clicked',
-        });
-      },
-      onRemove: (id: string) => {
-        unregister?.(id);
-        if (getPinnedLogsCount() < PINNED_LOGS_LIMIT) {
-          setPinLineButtonTooltipTitle('Pin to content outline');
-        }
-        reportInteraction('explore_toolbar_contentoutline_clicked', {
-          item: 'section',
-          type: 'Logs:pinned:pinned-log-deleted',
-        });
-      },
-    });
+    const alreadyPinned = pinnedLogs?.find((pin) => pin === row.rowId);
+    if (alreadyPinned && row.rowId && allowUnPin) {
+      unregister?.(row.rowId);
+      reportInteraction('explore_toolbar_contentoutline_clicked', {
+        item: 'section',
+        type: 'Logs:pinned:pinned-log-deleted',
+      });
+    } else if (getPinnedLogsCount() !== PINNED_LOGS_LIMIT && !alreadyPinned) {
+      register?.({
+        id: row.rowId,
+        icon: 'gf-logs',
+        title: PINNED_LOGS_TITLE,
+        panelId: PINNED_LOGS_PANELID,
+        level: 'child',
+        ref: null,
+        color: LogLevelColor[row.logLevel],
+        childOnTop: true,
+        onClick: () => {
+          onOpenContext(row, () => {});
+          reportInteraction('explore_toolbar_contentoutline_clicked', {
+            item: 'section',
+            type: 'Logs:pinned:pinned-log-clicked',
+          });
+        },
+        onRemove: (id: string) => {
+          unregister?.(id);
+          reportInteraction('explore_toolbar_contentoutline_clicked', {
+            item: 'section',
+            type: 'Logs:pinned:pinned-log-deleted',
+          });
+        },
+      });
+
+      reportInteraction('explore_toolbar_contentoutline_clicked', {
+        item: 'section',
+        type: 'Logs:pinned:pinned-log-added',
```

Comment:
There's one more `reportInteraction(... type: 'Logs:pinned:pinned-log-added',`) below causing it to be tirggered twice (on pin and unpin). 

Also since we now unpin in two places it would be nice to split the logic into to separate functions to improve readability, wdyt?

Reply:
Yeah good catch, I refactored all of the `explore_toolbar_contentoutline_clicked` into a new file and removed that duplicate event. What do ya think?

Reply:
Putting this into a new file is nice and should prevent duplication in future. However, the ones you refactored are specific to logs only. There's 3 more `explore_toolbar_contentoutline_clicked` in `ContentOutline.tsx` and 1 in `Explore.tsx`. Given that the new file is put under content outline and is named ContentOutlineAnalyticEvents.ts we should probably refactor those too. @ifrost maybe we can do the remaining ones in a follow-up PR?